### PR TITLE
[Feat](json) Support json_search function

### PR DESCRIPTION
**Pull request by liutang123**
## Proposed changes
Like mysql, json_search returns the path witch point to a json string witch match the pattern.
`SELECT JSON_SEARCH('["A",[{"B":"1"}],{"C":"AB"},{"D":"BC"}]', 'one', 'A_') as res;`
```
+----------+
| res      |
+----------+
| "$[2].C" |
+----------+
```
Issue Number: close #xxx

### CODE REVIEW

**Comment by liutang123** [comment]
will add doc after pipeline is passed.

File: gensrc/script/doris_builtins_functions.py [hunk 1]
```diff
@@ -2292,7 +2294,8 @@
     'coalesce',
     'array',
     'json_array',
-    'json_object'
+    'json_object',
+    'json_search',
```
Comment (cambyzju) [inline gensrc/script/doris_builtins_functions.py:2298]:
```suggestion
    'json_search'
```

Comment (cambyzju) [inline gensrc/script/doris_builtins_functions.py:2298]: json_search returns ALWAYS_NULLABLE, not need to change 'non_null_result' vector.

File: be/src/vec/functions/function_jsonb.cpp [hunk 2]
```diff
@@ -1598,6 +1600,377 @@ struct JsonbContainsAndPathImpl {
...
+        auto result_col_nullable =
+                ColumnNullable::create(std::move(result_col), std::move(null_map));
+        block.replace_by_position(result, std::move(result_col_nullable));
+        return Status::OK();
+    }
+
+    static constexpr auto one = "one";
+    static constexpr auto all = "all";
+    static bool _is_one(const std::string& all_or_one) {
+        if (all_or_one == one) {
```
Comment (cambyzju) [inline be/src/vec/functions/function_jsonb.cpp:1799]: return strcasecmp(all_or_one.c_str(), "one") == 0;

File: be/src/vec/functions/function_jsonb.cpp [hunk 3]
```diff
@@ -1598,6 +1600,377 @@ struct JsonbContainsAndPathImpl {
...
+               (all_or_one.at(1) == 'l' || all_or_one.at(1) == 'L') &&
+               (all_or_one.at(2) == 'l' || all_or_one.at(2) == 'L');
+    }
+
+public:
+    static constexpr auto name = "json_search";
+    static FunctionPtr create() { return std::make_shared<FunctionJsonSearch>(); }
+
+    String get_name() const override { return name; }
+    bool is_variadic() const override { return true; }
```
Comment (cambyzju) [inline be/src/vec/functions/function_jsonb.cpp:1826]:
```suggestion
    bool is_variadic() const override { return false; }
```

**Comment by liutang123** [comment]
run buildall

**Comment by liutang123** [comment]
run buildall

**Review by cambyzju — DISMISSED** [review]
LGTM

**Comment by liutang123** [comment]
run buildall

**Review by cambyzju — APPROVED** [review]

**Review by lide-reed — APPROVED** [review]
LGTM

**Pull request merged**